godtools/AppDelegateExtensions/AppDelegate+UniversalLinks.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  AppDelegate+UniversalLinks.swift
+//  godtools
+//
+//  Created by Ryan Carlson on 7/14/17.
+//  Copyright © 2017 Cru. All rights reserved.
+//
+
+import Foundation
+import PromiseKit
+
+extension AppDelegate {
+    
+    func application(_ application: UIApplication, continue userActivity: NSUserActivity, restorationHandler: @escaping ([Any]?) -> Void) -> Bool {
+        if userActivity.activityType != NSUserActivityTypeBrowsingWeb {
+            return false
+        }
+        
+        guard let url = userActivity.webpageURL else {
+            return false
+        }
+        
```

Comment:
Shouldn't be good to check here `if pathParts.count < 3 ` in order to avoid the conditions in the following methods?

Reply:
Actually, I think it's better w/o that. The individual methods that check for resource, language, page in the URL check to make sure that the part of the path it's looking for exists before it accesses it. I change put in this statement here, but I think it will be more confusing and redundant.

````swift
        guard let url = userActivity.webpageURL, url.pathComponents.count >= 4 else {
            return false
        }
````